presto-main/src/main/java/io/prestosql/server/TaskResource.java

```diff
@@ -110,9 +111,17 @@ public TaskResource(
     @ResourceSecurity(INTERNAL_ONLY)
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
+    public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo, @QueryParam("state") String stateFilter)
```

Comment:
nit: you probably can `String stateFilter` -> `@Nullable TaskState stateFilter`  or even `Optional<TaskState> stateFilter`